src/v/raft/group_configuration.h

```diff
@@ -57,25 +53,45 @@ class vnode : public serde::envelope<vnode, serde::version<0>> {
     model::revision_id _revision;
 };
 
-enum class configuration_type : uint8_t { simple, joint };
+enum class configuration_type : uint8_t { simple, joint, transitional };
```

Comment:
Looks like this enum becomes less of a "type" (a stateless property), and more of a "state/step" in the reconfiguration transition. So maybe a rename is needed + some additional comments describing the possible transitions.

Reply:
done, good idea